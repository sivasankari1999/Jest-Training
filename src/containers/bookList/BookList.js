import React from 'react'
import BookComp from '../../components/bookComp/BookComp'

export class BookList extends React.Component {
    state={
        booklist:{
            book1:"PonniyinSelvan-I",
            book2:"Harry Potter and Deathly Hollows",
            book3:"Harry Potter and the chamber of secrets",
            book4:"Harry Potter and the half Blood Prince"
        }
    }
  render() {
    return (
      <div>
          <h3> BookList Component </h3>
          <p data-testid="heading">Wel­come to My Fa­vorite Book Por­tal</p>
           <p>{this.state.booklist.book1}</p> 
           <p>{this.state.booklist.book2}</p> 
           <p>{this.state.booklist.book3}</p> 
           <p>{this.state.booklist.book4}</p> 
          <br/>
          <br/>
          <br/>
          <BookComp bookName={this.state.booklist}/>
      </div>
    )
  }
}

export default BookList