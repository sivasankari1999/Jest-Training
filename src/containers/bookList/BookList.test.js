
import { render,screen } from "@testing-library/react";
import BookList from "./BookList";
import renderer from 'react-test-renderer';
describe("bookList Component",()=>{
    it("should have the text ",()=>{
        render(<BookList/>);
        const headingEl = screen.getByTestId('heading');
        expect(headingEl).toHaveTextContent('Wel­come to My Fa­vorite Book Por­tal');
    })
    it('should have right booklist snapshot',()=>{
        const snapshotJSON = renderer.create(<BookList />).toJSON();
        expect(snapshotJSON).toMatchSnapshot();
    })
})

