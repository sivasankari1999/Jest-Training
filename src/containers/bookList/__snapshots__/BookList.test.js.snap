// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookList Component should have right booklist snapshot 1`] = `
<div>
  <h3>
     BookList Component 
  </h3>
  <p
    data-testid="heading"
  >
    Wel足come to My Fa足vorite Book Por足tal
  </p>
  <p>
    PonniyinSelvan-I
  </p>
  <p>
    Harry Potter and Deathly Hollows
  </p>
  <p>
    Harry Potter and the chamber of secrets
  </p>
  <p>
    Harry Potter and the half Blood Prince
  </p>
  <br />
  <br />
  <br />
  <div>
    <h3>
      Book Component
    </h3>
    <p
      data-testid="booklist1"
    >
      PonniyinSelvan-I
    </p>
    <p
      data-testid="booklist2"
    >
      Harry Potter and Deathly Hollows
    </p>
    <p
      data-testid="booklist3"
    >
      Harry Potter and the chamber of secrets
    </p>
    <p
      data-testid="booklist4"
    >
      Harry Potter and the half Blood Prince
    </p>
    <br />
    <button
      data-testid="Readingbtn"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "lightblue",
          "margin": "20px 40px",
          "padding": "10px 20px",
        }
      }
    >
      Add to Read足ing List
    </button>
  </div>
</div>
`;
